specs/marketplace.md

```diff
@@ -14,58 +14,52 @@ contributors:
+| Slot or Storage Slot       | A space allocated by the Storage Request to store a single chunk of the data related to this Storage Request.              |
+| Smart Contract             | A smart contract associated with marketplace functionality.                                                               |
+| Codex Tokens               | ERC20-based tokens used within the Codex network.                                                                          |
 
 ## Motivation
 
-The Codex network aims to create a peer-to-peer storage engine with strong data durability,
-data persistence guarantees, and node storage incentives.
+The Codex network aims to create a peer-to-peer storage engine with robust data durability, data persistence guarantees, and incentives for node storage.
 
-An important component of the Codex network is its Marketplace. It is a place which mediates negotiations of all parties
- to provide persistence in the network. It also provides ways to enforce agreements and facilitate repair when a storage node drops out.
+The marketplace is a critical component of the Codex network, serving as a platform where all involved parties interact to ensure data persistence. It also provides mechanisms to enforce agreements and facilitate data repair when Storage Nodes fail to fulfill their duties.
 
-The marketplace is defined by a smart contract deployed to an EVM-compatible blockchain. It has several flows which are 
-linked with roles in the network and which the participating node takes upon itself. Each node can take on the responsibilities of one role or multiple at the same time.
-This specification describes these flows.
+Implemented as a Smart Contract on an EVM-compatible blockchain, the Marketplace enables various scenarios where nodes assume one or more roles to maintain a reliable persistence layer for users. This specification details these interactions.
 
-The Marketplace handles storage requests, the storage slot state,
-storage provider rewards, storage provider collaterals, and storage proof state.
+The Marketplace contract manages Storage Requests, maintains the state of allocated Storage Slots, and orchestrates Storage Provider rewards, collaterals, and storage proofs.
 
-If a node implementation wants to participate in the persistence layer of Codex, it needs to choose which role(s) it wants
-to support and implement proper flows. Otherwise, it won't be compatible with the rest of the Codex network.
+A node that wishes to participate in the Codex persistence layer MUST implement one or more roles described in this document.
 
 ### Roles
 
-There are three main roles in the network: client, storage provider (SP) and validator. 
+A node can assume one of the three main roles in the network: the Client, the Storage Provider (SP), and the Validator (for the sake of conciseness and where it does not cause ambiguity, referred to as a Client, Storage Provider, and Validator, respectively).
```

Comment:
@AuHau, yes, just my comment - You saw it so I can remove it now.

Reply:
In the original spec those sometimes were capitalised and sometimes not. I saw that practice of capitalising things that have special meaning in the given context. I am fine to have it all lowercase. Just trying to have some logic behind. For instance: a request. We have a type called a Request and that (I think) was a reason why sometimes in the original spec Adam keept it in uppercase, and because somehow I felt it a Request is too wide, so in some places I changed it to Storage Request. It is also widely observed that they can make reading the spec harder. Maybe a good approach is to keep it all lowercase for now as you suggest, and only later we can decide if some instances must be changed.

Reply:
Last couple of thoughts:

- maybe to remove ambiguity, we should always say `storage request` where we say `request` even it can be figured out from the context
- Smart Contract is not just a smart contract. It is THE SMART CONTRACT that captures the marketplace functionality. How is it different (in terms of being a proper name) from "Token"?
- When we are saying "Marketplace" (so capitalised), it is kind of more clear to me we are talking about "Codex Marketplace". When we just say "a marketplace", or "the marketplace" it feels to me less related to Codex Marketplace.

Fortunately it is super easy to change the caps....